packages/circuit-ui/components/Popover/Popover.tsx

```diff
@@ -300,80 +324,47 @@ export const Popover = ({
-            position: 'fixed',
-            zIndex: (zIndex || theme.zIndex.popover).toString(),
-          };
-        } else {
-          // eslint-disable-next-line no-param-reassign
-          state.styles.popper.width = 'auto';
-          // eslint-disable-next-line no-param-reassign
-          state.styles.popper.zIndex = (
-            zIndex || theme.zIndex.popover
-          ).toString();
-        }
-      },
-    }),
-    [theme, zIndex],
-  );
-
-  // Flip the popover if there is no space for the default placement.
-  // https://popper.js.org/docs/v2/modifiers/flip/
-  const flip = {
-    name: 'flip',
-    options: {
-      fallbackPlacements,
-    },
+  const handleTriggerClick = (event: ClickEvent) => {
+    // This prevents the event from bubbling which would trigger the
+    // useClickOutside above and would prevent the popover from closing.
+    event.stopPropagation();
+    handleToggle((prev) => !prev);
   };
 
-  // Disable the scroll and resize listeners when the popover is closed.
-  // https://popper.js.org/docs/v2/modifiers/event-listeners/
-  const eventListeners = {
-    name: 'eventListeners',
-    options: { scroll: isOpen, resize: isOpen },
+  const handleTriggerKeyDown = (event: KeyboardEvent) => {
+    if (isArrowDown(event)) {
+      triggerKey.current = 'ArrowDown';
+      handleToggle(true);
+    }
+    if (isArrowUp(event)) {
+      triggerKey.current = 'ArrowUp';
+      handleToggle((prev) => !prev);
+    }
   };
 
-  // Note: the usePopper hook intentionally takes a DOM node, not a ref, in
-  // order to update when the node changes. We use a callback ref for this.
-  const [popperElement, setPopperElement] = useState<HTMLElement | null>(null);
-  const { styles, attributes, update } = usePopper(
-    triggerEl.current,
-    popperElement,
-    {
-      strategy: 'fixed',
-      placement,
-      modifiers: [mobilePosition, flip, eventListeners, ...modifiers],
-    },
-  );
-
-  // This is a performance optimization to prevent event listeners from being
-  // re-attached on every render.
-  const popperRef = useLatest(popperElement);
+  const handlePopoverItemClick = (
+    event: ClickEvent,
+    onClick: BaseProps['onClick'],
+  ) => {
+    if (onClick) {
+      onClick(event);
+    }
+    handleToggle(false);
+  };
 
   useEscapeKey(() => handleToggle(false), isOpen);
-  useClickOutside(popperRef, () => handleToggle(false), isOpen);
-
-  const prevOpen = usePrevious(isOpen);
+  useClickOutside(floatingRef, () => handleToggle(false), isOpen);
 
   useEffect(() => {
-    if (update) {
-      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    // When the floating element is visible, add event listeners that invoke the function `update`
+    // to update the position of the floating element when screen is resized or scrolled
+    if (isOpen) {
       update();
+      window.addEventListener('resize', update);
+      window.addEventListener('scroll', update, { passive: true });
```

Comment:
Maybe we should look into throttling these for better performance. But this could also be a follow-up PR 🙂

Reply:
Would love to address this in a follow-up PR :)